test/test_ir.py

```diff
@@ -1,16 +1,81 @@
 from __future__ import annotations
 
 import math
+from collections.abc import Sequence
+from typing import Any
 
 import numpy as np
 import pytest
+from numpy.typing import ArrayLike
 
 from opensquirrel.common import ATOL
-from opensquirrel.ir import BlochSphereRotation, ControlledGate, MatrixGate, Measure, Qubit
+from opensquirrel.ir import Axis, BlochSphereRotation, ControlledGate, Expression, MatrixGate, Measure, Qubit
+
+
+class TestAxis:
+
+    @pytest.fixture(name="axis")
+    def axis_fixture(self) -> Axis:
+        return Axis(1, 0, 0)
+
+    @pytest.mark.parametrize("expected_class", [Sequence, Expression])
+    def test_inheritance(self, axis: Axis, expected_class: type[Any]) -> None:
+        assert isinstance(axis, expected_class)
+
+    def test_axis_getter(self, axis: Axis) -> None:
+        np.testing.assert_array_equal(axis.axis, [1, 0, 0])
+
+    @pytest.mark.parametrize(
+        "new_axis, expected_axis", [([0, 0, 1], [0, 0, 1]), ([0, 3, 4], [0, 3 / 5, 4 / 5]), (Axis(0, 1, 0), [0, 1, 0])]
```

Comment:
```suggestion
        "new_axis, expected_axis", [
            ([0, 0, 1], [0, 0, 1]),
            ([0, 3, 4], [0, 3 / 5, 4 / 5]),
            (Axis(0, 1, 0), [0, 1, 0])
        ]
```

Just a suggestion, for readability, a few lines following the pattern `(new_axis, expected_axis)`.

Reply:
I agree, done